.github/workflows/job-checks.yml

```diff
@@ -16,10 +16,19 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
           enable-cache: true
+
+      - name: pre-commit cache key
+        run: echo "PY=$(python -VV | sha256sum | cut -d' ' -f1)" >> "$GITHUB_ENV"
+      - uses: actions/cache@v4
+        with:
+          path: ~/.cache/pre-commit
+          key: pre-commit|${{ env.PY }}|${{ hashFiles('.pre-commit-config.yaml') }}
+
       - name: Install dependencies
-        run: uv sync --all-extras --no-extra cu124
+        run: uv sync --frozen --all-extras --no-extra cu124
+
       - name: Run styling check
-        run: uv run --no-sync pre-commit run --all-files
+        run: pre-commit run --all-files
```

Comment:
It looks like most projects are doing something like:

```
uv tool run --with pre-commit-uv pre-commit run ...
```

this could remove the need for the extra dev deps? or are there recommendations of when using it as dev deps vs as a tool?

Reply:
I use it locally. Dont know what works for other people

Reply:
It pulls `uv` inside venv as dev dep which might not be ideal